### PR TITLE
Multiset

### PR DESCRIPTION
**Pull request by simonwhitaker**
### Checklist

- [x] I've looked at the [contribution guidelines](https://github.com/raywenderlich/swift-algorithm-club/blob/master/.github/CONTRIBUTING.md).
- [x] This pull request is complete and ready for review.

### Description

This PR adds the Multiset (or "Bag") data structure; a set where the number of times an element is added matters.

### CODE REVIEW

File: Multiset/README.markdown [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+}
+```
+
+Getting the count for an item is simple: we just return the value for the given item in the internal dictionary.
+
+``` swift
+public func count(for key: Element) -> UInt {
+  return storage[key] ?? 0
+}
+```
```
Comment (kelvinlauKL) [inline Multiset/README.markdown]: It is customary to leave a footer crediting the contributors here (credit yourself!):

_Written for the Swift Algorithm Club by Simon Whitaker_

File: Multiset/Multiset.playground/Sources/Multiset.swift [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+//  Multiset.swift
+//  Multiset
+//
+//  Created by Simon Whitaker on 28/08/2017.
+//
+
+import Foundation
+
+public struct Multiset<Element: Hashable> {
+  fileprivate var storage: [Element: UInt]
```
[the remark this answers is not in the document]
Comment (simonwhitaker) [reply Multiset/Multiset.playground/Sources/Multiset.swift]: 👍

File: Multiset/Multiset.playground/Sources/Multiset.swift [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+  private var _count: UInt = 0
```
[the remark this answers is not in the document]
Comment (simonwhitaker) [reply Multiset/Multiset.playground/Sources/Multiset.swift]: Ah, nice. I didn't know that one. 😄

**Pull request merged**